tests/node.lisp

```diff
@@ -180,7 +180,13 @@ Finally, all of the nodes constructed by this BLOSSOM-LET are stashed in the pla
   (let (addresses augmented-node-definitions)
     ;; precompute addresses for the nodes
     (loop :for (symbol-name . rest) :in node-definitions
+          ;; if the node arguments include :process-courier, use the provided
+          ;; courier instance when pre-registering the address for this node
+          :for courier-idx := (position ':process-courier rest)
+          :for courier := (when courier-idx (nth (1+ courier-idx) rest))
+          :for courier-clause := (when courier `(:courier ,courier))
           :collect `(,symbol-name (register
+                                   ,@courier-clause
                                    :channel ',(gensym
                                                (format nil "TEST-BLOSSOM-~A"
                                                        symbol-name))))
```

Comment:
An alternative which combines the two `loop`s:

```lisp
    (loop :for (symbol-name . rest) :in node-definitions
          :for class-name := (if (keywordp (first rest))
                                 'blossom-node
                                 (first rest))
          :for chopped-args := (if (keywordp (first rest))
                                   rest
                                   (rest rest))
          ;; precompute addresses for the nodes.
          ;; if the node arguments include :process-courier, use the provided
          ;; courier instance when pre-registering the address for this node
          :for courier := (getf chopped-args :process-courier)
          :for courier-clause := (when courier `(:courier ,courier))
          :collect `(,symbol-name (register
                                   ,@courier-clause
                                   :channel ',(gensym
                                               (format nil "TEST-BLOSSOM-~A"
                                                       symbol-name))))
            :into addresses-result
          ;; then, set up definitions from clauses
          :collect `(,symbol-name (spawn-process ',class-name
                                                 :process-key ,symbol-name
                                                 ,@chopped-args
                                                 ,@global-options
                                                 :debug? t))
            :into definitions-result
          :finally (setf addresses addresses-result
                         augmented-node-definitions definitions-result))
```

Reply:
Sweet! Made the change